python/lsst/daf/butler/registry/wildcards.py

```diff
@@ -490,6 +490,11 @@ def require_ordered(self) -> tuple[str, ...]:
             )
         return self.strings
 
+    def empty(self) -> bool:
+        """Return true if both ``strings`` and ``patterns`` are empty."""
+        # bool(Ellipsis) is True
+        return not self.strings and not self.patterns
+
```

Comment:
Would it make sense to implement `__bool__` instead, with the opposite sign?

Reply:
I thought about `__bool__`, but I decided that explicit method is better, it's too easy to mix check for True/False with check for `None`.